### ISSUE TITLE
NO_COLOR support is buggy

### ISSUE THREAD

**Issue by sigmaSd** [helix-editor/helix#8071]
### Summary

there are a couple of bugs I noticed when I use NO_COLOR

- the cursor disappears sometimes
- there are still colors for example the swigly lines under errors (coming from the lsp)
- the selection mode have no indication (maybe thats not possible in no_color ?)

### Reproduction Steps

- create a rust project
- NO_COLOR=1 hx src/main.rs

expected: cursor visible
actual: cursor invisible

- add an extra `z` after println like `println!("Hello, world!")z;`
expected: swigly lines under z have no color
actual: swigly lines under z is red

- hit `v` to go to visual mode
expected there is some indication of what I'm selecting
actual: no indication of what I'm selecting

### Platform

linux

### Terminal Emulator

wezterm 20230712-072601-f4abf8fd

### Helix Version

helix 23.05 (c9694f68)

**Comment by kirawi** [comment]
I don't think it's supported: https://github.com/helix-editor/helix/issues/1923

**Comment by pascalkuthe** [comment]
Yeah that enviorment variable is part of a recent release. I don't think we want to support that. Running an interactive editor without colors makes no sense (for example the selection is rendered using clbg styling so without colours support cursors and selection are invisible)
